ag_data/formulas/tests.py

```diff
@@ -66,6 +82,39 @@ def test_save_measurement_identity(self):
         self.assertEqual(result.event_uuid, self.event)
         self.assertEqual(result.sensor_id, sensor)
 
+    def test_no_event(self):
+        formula_id = get_formula_id(formulas.flow_sensor)
+        sensor_type = AGSensorType.objects.filter(processing_formula=formula_id).first()
+        sensor = AGSensor.objects.create(name="Test Sensor", type_id=sensor_type)
+
+        backup_ag_active_event = AGActiveEvent.objects.all()
+        backup_event = self.pipeline.event
+
+        AGActiveEvent.objects.all().delete()
+        self.pipeline.event = None
+
+        with self.assertRaisesMessage(Exception, "No active event"):
+            self.pipeline.save_measurement(
+                sensor, self.timestamp, {"volumetricFlow": 2.0}
+            )
+
+        error_data = {
+            "sensor": sensor,
+            "timestamp": self.timestamp,
+            "measurement": {"volumetricFlow": 2.0},
+        }
+        self.assertTrue(
+            self.assert_error(
+                error_code=record.ERROR_CODE["NO_ACT_EVENT"],
+                description="Currently no active event",
+                raw_data=str(error_data),
+            )
+        )
+
+        for act_event_obj in backup_ag_active_event:
+            act_event_obj.save()
+        self.pipeline.event = backup_event
```

Comment:
I'm not 100% sure but I guess you don't need these 3 lines. `unittest` framework will call `setUp()` for every testcases so you don't need to restore `self.pipeline` by yourself. And it'll do database rollback as well so that you don't have to restore `act_event_obj` models.

Could you delete these lines if there's no problem without them?

Reply:
Done